### ISSUE TITLE
Have github actions quit completely if the docker build steps (or render steps) fail

### ISSUE THREAD

**Issue by cansavvy** [AlexsLemonade/refinebio-examples#312]
https://github.com/AlexsLemonade/refinebio-examples/runs/1265424176?check_suite_focus=true

I'm not sure why this is. I can push the image manually, but its odd the Docker image with `clusterProfiler` didn't make it to Dockerhub. 

It's been failing periodically because of this.

**Comment by cansavvy** [comment]
So the build failed because of a dependency but it didn't actually stop the build it just kept moving on anyway. I didn't expect this. 

```
ERROR: dependency ‘enrichplot’ is not available for package ‘clusterProfiler’
```

**Comment by cansavvy** [comment]
I don't like that the renders have been failing this whole time but it didn't seem to tell us. This seems like not what we want for our processes. Can we make this rendering completion a necessary check? Looks like currently it is not required (not sure why it's not).

**Comment by cansavvy** [comment]
To be clear, I don't *think* this is a fail to push problem, I think this was a weird instance that something failed to install but the docker kept building anyway, which is why we didn't see an error until render. The error I posted above was buried in a lot of docker output text but if you look at that instance, it didn't fail the docker build with that error even though the packages didn't install.

**Comment by jaclyn-taroni** [comment]
Change the title to reflect what needs to be solved now that you have an idea.

**Comment by jashapiro** [comment]
I guess I assumed that getting an exit code 1 along the way would fail the a github actions step, but clearly not. We should be able to make that happen

**Comment by jashapiro** [comment]
There seem to be 2 issues here: Dockerfile fix and GHA fix.

**Comment by cansavvy** [comment]
> There seem to be 2 issues here: Dockerfile fix and GHA fix.

Dockerfile fix is easy, [this works](https://github.com/AlexsLemonade/refinebio-examples/issues/312#issuecomment-713025216). GHA fix is really what we are left with.

**Comment by jaclyn-taroni** [comment]
> I guess I assumed that getting an exit code 1 along the way would fail the a github actions step, but clearly not. We should be able to make that happen

There is an exit code 1 for the render step here: https://github.com/AlexsLemonade/refinebio-examples/runs/1265424176?check_suite_focus=true

So I take it that getting an exit code 1 _does_ fail the GHA step.

> > There seem to be 2 issues here: Dockerfile fix and GHA fix.
> 
> Dockerfile fix is easy, [this works](https://github.com/AlexsLemonade/refinebio-examples/issues/312#issuecomment-713025216). GHA fix is really what we are left with.


That solves the Dockerfile problem for _an individual dependency_, but why didn't we get a failing exit code for the Docker build step?

Where does the error message in https://github.com/AlexsLemonade/refinebio-examples/issues/312#issuecomment-713025216 come from?

**Comment by cansavvy** [comment]
> There is an exit code 1 for the render step here: https://github.com/AlexsLemonade/refinebio-examples/runs/1265424176?check_suite_focus=true
> 
> So I take it that getting an exit code 1 _does_ fail the GHA step.

Right. But it still got through `docker-build` (not to be confused with `docker-build-push`) without failing. 

> Where does the error message in [#312 (comment)](https://github.com/AlexsLemonade/refinebio-examples/issues/312#issuecomment-713025216) come from?

I had to look in the output for the docker build step in that [same gha report](https://github.com/AlexsLemonade/refinebio-examples/runs/1265424176?check_suite_focus=true) and search for `clusterProfiler` which brought me to the error that docker didn't stop for (which is perplexing in itself that it didn't stop when it failed to install). 


> 
> That solves the Dockerfile problem for _an individual dependency_, but why didn't we get a failing exit code for the Docker build step?

Docker didn't give me an error locally either. It kept rolling even though it didn't install (and I didn't know it).

**Comment by jaclyn-taroni** [comment]
Yes, so the Dockerfile problem is we are getting a warning, not an _error_. This is sometimes a problem when installing Bioconductor packages on a Docker image (or at least it was in the `biocLite()` days). So we can try setting `options(warn = 2)` _somewhere_ (see: https://github.com/AlexsLemonade/refinebio/blob/6da85928b63537ddda272ff8fb4d79e2235209b1/workers/affymetrix_dependencies.R#L3). Maybe something like `options(warn = 2); <command for installing Bioconductor packages>`.

**Comment by cansavvy** [comment]
This is resolved by #316, the related issue #315 is still at large.
